### ISSUE TITLE
[Bug]: failed to locate the violation line for sql-check

### ISSUE THREAD

**Issue by yhilmare** [oceanbase/odc#1303]
### ODC version

4.2.3

### OB version

not related

### What happened?

1. type several sqls:
```SQL
CREATE TABLE range_parti_global_tbl (
  name varchar2(64) DEFAULT NULL,
  age integer DEFAULT NULL,
  UNIQUE (age) using index global
) partition by range (age)
(partition p1 values less than (5),
partition p2 values less than (10));

alter table range_parti_global_tbl drop partition p1 update global indexes;

select * from range_parti_global_tbl;

show variables like '%version%';

insert into range_parti_global_tbl values('David', 3), ('Marry', 8);
```
2. click 'sql-check' and locate the violation:
<img width="1048" alt="截屏2023-12-25 17 13 53" src="https://github.com/oceanbase/odc/assets/22992296/49033d8d-82a6-48d2-8e23-a38abe0119cb">

as you can see, the location is wrong.

### What did you expect to happen?

locate the violation successfully.

### How can we reproduce it (as minimally and precisely as possible)?

follow what happened

**Comment by HSunboy** [comment]
Fixed In Commit [a6acfbe30c4d9694d35d27ce39c940a02f74410f](https://github.com/oceanbase/odc-client/commit/a6acfbe30c4d9694d35d27ce39c940a02f74410f)
            ------
            陈朝俊
            Mon Dec 25 18:27:31 2023 +0800

**Comment by tonghan-gsh** [comment]
![Image](https://github.com/oceanbase/odc/assets/46598871/25576e95-51c4-4b3e-ab62-9958c7a35960)

pass
